CMakeLists.txt

```diff
@@ -501,21 +501,20 @@ if(${use_http})
         set(aziotsharedutil_target_libs ${aziotsharedutil_target_libs} crypt32 winhttp)
     else()
         if (NOT use_builtin_httpapi)
-            if (CMAKE_CROSSCOMPILING)
-                # As mentioned at https://cmake.org/Wiki/CMake_Cross_Compiling the
-                # pkg-config tool can not be used by cmake while cross compiling.
-                message(STATUS "Cross compiling not using pkg-config")
-            else()
-                # try pkg-config first
-                find_package(PkgConfig)
-                if(PKG_CONFIG_FOUND)
-                    pkg_check_modules(CURL libcurl)
-                endif()
-            endif()
 
-            # if that didn't work, try CMake find_package
-            if(NOT CURL_FOUND)
-                find_package(CURL)
+            # try CMake find_package first
+            find_package(CURL)
```

Comment:
I hope this will not result in failure for some platform that 1) is not WIN32, AND 2) is not using buit-in http API. In such case we are requiring CURL to be present. But anyway, that was the behavior before.

Reply:
This was in case it was not found using pkg-config, to try another way, and only for cross-compilation.  the logic is still there for cross-compilation (just reversed cmake to be used first, then pkg-config)